sql/create.go

```diff
@@ -142,6 +142,8 @@ func (p *planner) CreateIndex(n *parser.CreateIndex) (planNode, error) {
 		return nil, err
 	}
 
+	// Mark transaction as operating on the system DB.
+	p.txn.SetSystemDBTrigger()
```

Comment:
The way we tell the transaction to add a trigger changed, but explicitly setting it stays.